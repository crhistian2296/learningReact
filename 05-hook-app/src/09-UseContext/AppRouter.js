import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LogginScreen from './LogginScreen';

const AppRouter = () => (
  <>
    <Router>
      <Routes>
        <Route path='/about' element={<AboutScreen />} />
        <Route path='/loggin' element={<LogginScreen />} />
        <Route path='/' element={<HomeScreen />} />
      </Routes>
    </Router>
  </>
);
export default AppRouter;
