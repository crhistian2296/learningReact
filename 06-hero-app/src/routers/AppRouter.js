import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import LoginScreen from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

const AppRouter = () => {
  return (
    <Router>
      <div className='main'>
        <Routes>
          {/* <Route path='login' element={<LoginScreen />} /> */}
          <Route
            path='login'
            element={
              <PublicRoutes>
                <LoginScreen />
              </PublicRoutes>
            }
          />
          <Route
            path='*'
            element={
              <PrivateRoutes>
                <DashboardRoutes />
              </PrivateRoutes>
            }
          />
        </Routes>
      </div>
    </Router>
  );

  // *react router v6 convention
  // return (
  //   <Router>
  //     <div className='main'>
  //       <Routes>
  //         <Route path='login' element={<LoginScreen />} />
  //         <Route path='' element={<PrivateRoutes />}>
  //           <Route element={<DashboardRoutes />}>
  //             <Route path='hero/:heroId' element={<HeroScreen />} />
  //             <Route path='marvel' element={<MarvelScreen />} />
  //             <Route path='dc' element={<DCScreen />} />
  //             <Route path='search' element={<SearchScreen />} />

  //             {/* Default */}
  //             <Route path='/' element={<MarvelScreen />} />
  //           </Route>
  //         </Route>
  //       </Routes>
  //     </div>
  //   </Router>
  // );
};

export default AppRouter;
