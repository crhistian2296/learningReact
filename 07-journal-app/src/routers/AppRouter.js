import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { BrowserRouter as Router, Switch } from 'react-router-dom';
import { login } from '../actions/auth';
import { startLoadingNotes } from '../actions/notes';
import JournalScreen from '../components/journal/JournalScreen';
import { firebase } from '../firebase/firebase-config';
import AuthRouter from './AuthRouter';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

const AppRouter = () => {
  const [checking, setChecking] = useState(true);
  const [loggedIn, setLoggedIn] = useState(false);
  const dispatch = useDispatch();

  useEffect(() => {
    firebase.auth().onAuthStateChanged(async (user) => {
      if (user?.uid) {
        dispatch(login(user.uid, user.displayName));
        setLoggedIn(true);

        dispatch(startLoadingNotes(user.uid));
      } else setLoggedIn(false);
      setChecking(false);
    });
  }, [dispatch, setChecking]);

  if (checking) return <h1>Wait...</h1>;

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoutes path='/auth' component={AuthRouter} isAuthenticated={loggedIn} />
          <PrivateRoutes exact path='/' component={JournalScreen} isAuthenticated={loggedIn} />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
