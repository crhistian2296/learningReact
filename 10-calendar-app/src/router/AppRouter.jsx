import { Navigate, Route, Routes } from 'react-router-dom';
import { LoginPage } from '../auth';
import { CalendarPage } from '../calendar/pages/CalendarPage';

const AppRouter = () => {
  const authStatus = 'authenticated'; // 'authenticated' | 'not-authenticated'
  return (
    <>
      <Routes>
        {authStatus === 'not-authenticated' ? (
          <Route path='/auth/*' element={<LoginPage />}></Route>
        ) : (
          <Route path='/*' element={<CalendarPage />}></Route>
        )}
        <Route path='/*' element={<Navigate to={'/auth/login'} />}></Route>
      </Routes>
    </>
  );
};

export default AppRouter;
