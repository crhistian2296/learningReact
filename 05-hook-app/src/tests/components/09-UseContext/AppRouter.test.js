import { mount } from 'enzyme';
import AppRouter from '../../../09-UseContext/AppRouter';

describe('Pruebas en <AppRouter/>', () => {
  const wrapper = mount(<AppRouter />);

  test('Debe de renderizar correctamente ', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
