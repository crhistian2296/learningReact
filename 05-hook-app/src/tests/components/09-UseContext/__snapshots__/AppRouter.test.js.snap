// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Debe de renderizar correctamente  1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-item nav-link  active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-item nav-link "
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-item nav-link "
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
    </div>
  </div>,
]
`;
